security/json_login_setup.rst

```diff
@@ -0,0 +1,229 @@
+                ),
+            ),
+        ));
+
+.. tip::
+
+    The ``check_path`` can also be route names (but cannot have mandatory wildcards - e.g.
+    ``/login/{foo}`` where ``foo`` has no default value).
+
+Create a new ``SecurityController`` inside a bundle::
+
+    // src/AppBundle/Controller/SecurityController.php
+    namespace AppBundle\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    class SecurityController extends Controller
+    {
+    }
+
+Next, configure the route that you earlier used under your ``json_login``
+configuration (``login``):
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/AppBundle/Controller/SecurityController.php
+
+        // ...
+        use Symfony\Component\HttpFoundation\Request;
+        use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+        class SecurityController extends Controller
+        {
+            /**
+             * @Route("/login", name="login")
+             */
+            public function loginAction(Request $request)
+            {
+            }
+        }
+
+    .. code-block:: yaml
+
+        # app/config/routing.yml
+        login:
+            path:     /login
+            defaults: { _controller: AppBundle:Security:login }
+
+    .. code-block:: xml
+
+        <!-- app/config/routing.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                http://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="login" path="/login">
+                <default key="_controller">AppBundle:Security:login</default>
+            </route>
+        </routes>
+
+    ..  code-block:: php
+
+        // app/config/routing.php
+        use Symfony\Component\Routing\RouteCollection;
+        use Symfony\Component\Routing\Route;
+
+        $collection = new RouteCollection();
+        $collection->add('login', new Route('/login', array(
+            '_controller' => 'AppBundle:Security:login',
+        )));
+
+        return $collection;
+
+Great!
+
+Don't let this controller confuse you. As you'll see in a moment, when the
+user submits the form, the security system automatically handles the form
+submission for you. If the user submits an invalid username or password,
+this controller reads the form submission error from the security system,
+so that it can be displayed back to the user.
+
+In other words the security system itself takes care of checking the submitted
+username and password and authenticating the user.
+
+And that's it! When you submit a ``POST`` request to the ``/login`` URL with
+the following JSON document as body, the security system will automatically
+check the user's credentials and either authenticate the user or throw an error::
+
+.. code-block:: json
```

Comment:
For some reason, this block does not render well: http://pr-7081-acxnhhi-6qmocelev2lwe.eu.platform.sh/security/json_login_setup.html

(probably because of the double colon L.149)